src/WebInterface/Infrastructure/Security/ArrivalAuthenticator.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gaming\WebInterface\Infrastructure\Security;
+
+use Gaming\WebInterface\Application\IdentityService;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+use Symfony\Component\Security\Core\Authentication\Token\TokenInterface;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+use Symfony\Component\Security\Http\Authenticator\AbstractAuthenticator;
+use Symfony\Component\Security\Http\Authenticator\Passport\Badge\UserBadge;
+use Symfony\Component\Security\Http\Authenticator\Passport\Passport;
+use Symfony\Component\Security\Http\Authenticator\Passport\SelfValidatingPassport;
+
+final class ArrivalAuthenticator extends AbstractAuthenticator
+{
+    public function __construct(
+        private readonly IdentityService $identityService,
+        private readonly TokenStorageInterface $tokenStorage
+    ) {
+    }
+
+    public function supports(Request $request): ?bool
+    {
+        return $this->tokenStorage->getToken() === null;
```

Comment:
The following snippet would create the user only for write requests. For read requests, the `Security` class takes care of that by default `NilUuid` is used. The snippet works, but the user interface has to be adapted, because the status of the page is not updated for ajax requests. For example, in the lobby the opened game is only highlighted after a refresh. Either fix that, or add the comment that already was written in `AssignUserIdOnKernelRequest`.

```php
if ($request->isMethodSafe()) {
    return false;
}
```

Reply:
Created and linked issue #155.